axiom/tests/test_php_packages.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_php_packages.h"
+#include "tlib_main.h"
+#include "util_memory.h"
+
+static void test_php_package_create_destroy(void) {
+  nr_php_package_t* package;
+  nr_php_package_t* null_package = NULL;
+
+  // Test: create new package and ensure it contains correct information
+  package = nr_php_package_create("Laravel", "8.83.27");
+
+  tlib_pass_if_not_null("create package", package);
+  tlib_pass_if_str_equal("test package name", "Laravel", package->package_name);
+  tlib_pass_if_str_equal("test package version", "8.83.27",
+                         package->package_version);
+
+  nr_php_package_destroy(package);
+
+  // Test: passing NULL pointer should not cause crash
+  nr_php_package_destroy(null_package);
+  nr_php_package_destroy(NULL);
+}
+
+static void test_php_adding_packages_to_hashmap(void) {
+  nr_php_package_t* package1;
+  nr_php_package_t* package2;
+  nr_php_package_t* package3;
+  nr_php_packages_t* hm = nr_php_packages_create();
+  int count;
+
+  // Test: create multiple new packages and add to hashmap
+  package1 = nr_php_package_create("Package One", "10.1.0");
+  package2 = nr_php_package_create("Package Two", "11.2.0");
+  package3 = nr_php_package_create("Package Three", "12.3.0");
+  /* Should not crash: */
+  nr_php_packages_add_package(NULL, package1);
+  nr_php_packages_add_package(hm, NULL);
+  nr_php_packages_add_package(hm, package1);
+  nr_php_packages_add_package(hm, package2);
+  nr_php_packages_add_package(hm, package3);
+
+  count = nr_php_packages_count(hm);
+
+  tlib_pass_if_int_equal("package count", 3, count);
+
+  nr_php_packages_destroy(&hm);
+  tlib_pass_if_null("PHP packages hashmap destroyed", hm);
+}
+
+static void test_php_package_to_json(void) {
+  char* json;
+  nr_php_package_t* package1;
+
+  // Test: convert package to json
+  package1 = nr_php_package_create("TestPackage", "7.2.0");
+  json = nr_php_package_to_json(package1);
+  tlib_pass_if_str_equal("valid package", "[\"TestPackage\",\"7.2.0\",{}]",
+                         json);
+  nr_free(json);
+  nr_php_package_destroy(package1);
+}
+
+static void test_php_packages_to_json_buffer(void) {
```

Comment:
Test passing in a NULL pointer to `nr_php_packages_to_json_buffer()`.

Reply:
added: [2dc23ea](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/2dc23ea5ce9a0c5c0915502caf0396fce8002818)